runtime/registry.go

```diff
@@ -79,11 +72,148 @@ func (r *Runtime) CreateInstance(ctx context.Context, inst *drivers.Instance) er
 	return nil
 }
 
-func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string) error {
-	err := r.Registry().DeleteInstance(ctx, instanceID)
+func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string, dropDB bool) error {
```

Comment:
I don't feel good about the error handling in this function. In case of errors, the callers assumption must be that the delete failed and should be retried. So for example, returning an error because `r.catalogCache` didn't have an entry feels wrong.

I think it's worth going over this again with fault tolerance in mind. It might make sense to return certain errors, and ignore or log others.

I would also reconsider the `util.ReturnFirstErr`. Might not be needed given the above, but if it is, I'm not sure it's worth adding a package to save a few lines on a loop here.

Reply:
I will handle the case when instance is not found. The reason I was using `r.catalogCache` here was to get all the db handlers. I didn't want to copy all code around getting the catalog store. This should be removed with the refactors planned. With instance not found case handled, I think all other errors in creating a new catalog service instance (if no instance exists in cache already) should be retryable.